src/plugins/query_enhancements/public/query_assist/components/query_assist_summary.tsx

```diff
@@ -0,0 +1,286 @@
+        const hit: { [key: string]: any } = {};
+        data.fields.forEach((field) => {
+          hit[field.name] = field.values[index];
+        });
+        hits.push({
+          _index: data.name,
+          _source: hit,
+        });
+      }
+    }
+    return hits;
+  };
+
+  const { query, search } = props.data;
+
+  const reportMetric = props.usageCollection
+    ? (metric: string) => {
+        props.usageCollection.reportUiStats(
+          METRIC_APP,
+          props.usageCollection.METRIC_TYPE.CLICK,
+          metric + '-' + uuidv4()
+        );
+      }
+    : () => {};
+
+  const reportCountMetric = props.usageCollection
+    ? (metric: string, count: number) => {
+        props.usageCollection.reportUiStats(
+          METRIC_APP,
+          props.usageCollection.METRIC_TYPE.COUNT,
+          metric + '-' + uuidv4(),
+          count
+        );
+      }
+    : () => {};
+
+  useEffect(() => {
+    let dataStack = [];
+    const enum DataType {
+      QUESTION,
+      QUERY,
+      DATA,
+    }
+    const combineSubscription = merge(
+      question$.pipe(
+        filter((value) => !isEmpty(value)),
+        mergeMap((value) => of({ type: DataType.QUESTION, data: value }))
+      ),
+      query.queryString.query$.pipe(
+        filter((value) => !isEmpty(value)),
+        mergeMap((value) => of({ type: DataType.QUERY, data: value }))
+      ),
+      search.df?.df$?.pipe(
+        distinctUntilChanged(),
+        filter((value) => !isEmpty(value) && !isEmpty(value?.fields)),
+        mergeMap((value) => of({ type: DataType.DATA, data: value }))
+      )
+    ).subscribe((value) => {
+      // to ensure we only trigger summary when user hits the query assist button with natual language input
+      if (value.type === DataType.QUESTION) {
+        dataStack = [value.data];
+      } else if (value.type === DataType.QUERY && dataStack.length === 1) {
+        dataStack.push(value.data.query);
+      } else if (value.type === DataType.DATA && dataStack.length === 2) {
+        dataStack.push(value.data);
+        setQueryContext({
+          question: dataStack[0],
+          query: dataStack[1],
+          queryResults: convertResult(dataStack[2]),
+        });
+        dataStack = [];
+      }
+    });
+
+    return () => {
+      combineSubscription.unsubscribe();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    const fetchSummary = async () => {
+      if (isEmpty(queryContext?.queryResults)) return;
+      setLoading(true);
+      setSummary(null);
+      const SUCCESS_METRIC = 'fetch_summary_success';
+      try {
+        const actualSampleSize = Math.min(sampleSize, queryContext?.queryResults?.length);
+        const dataString = JSON.stringify(queryContext?.queryResults?.slice(0, actualSampleSize));
+        const payload = `'${dataString}'`;
+        const response = await props.http.post('/api/assistant/data2summary', {
```

Comment:
not familiar with `data2summary`, but i'm ok with it if our long term plan would be to combine query assist logic into dashboards-assistant plugin

Reply:
The API is in dashboard-assistant https://github.com/opensearch-project/dashboards-assistant/blob/main/server/routes/summary_routes.ts#L130

The plugin should be there, otherwise query assist feature won't be enabled

> if our long term plan would be to combine query assist logic into dashboards-assistant plugin

Glad to hear that we have this plan.